### PR TITLE
Add unit testing for ThreadChannel

### PR DESCRIPTION
**Pull request by egnor**
Also:
- update mocks to support that testing
- fix some bugs found by unit testing (yay)
- update some comments found along the way

### CODE REVIEW

**Comment by anaulin** [comment]
Hmm. I note that the fixes in commit https://github.com/EscapeRoomsCommunity/escape_roomba/pull/10/commits/87e2fa8017565f2efcfa73fcc42f01b3b3b77bc3 where not caught by the automated test run in CI. I guess those lines are not exercised anywhere in tests. Would be nice to improve that.

File: escape_roomba/thread_manager.py [hunk 1]
```diff
@@ -154,28 +151,30 @@ async def _async_message_update(self, channel_id=None, channel=None,
...
+                thread = self._thread_by_origin.get(ci, {}).get(mi)
+                if thread is not None:  # Message is an existing thread origin.
+                    await thread.async_refresh_origin()
+                    if thread.is_deleted:  # Thread deleted!
                         del self._thread_by_origin[ci][mi]
-                        del self._thread_by_channel[t.thread_channel.id]
+                        del self._thread_by_channel[thread.thread_channel.id]
                 else:  # No thread based on this message; maybe make one?
-                    t = await ThreadChannel.async_maybe_create_from_origin(
+                    thread = await ThreadChannel.async_maybe_create_from_origin(
```
Comment (anaulin) [inline escape_roomba/thread_manager.py:161]: You sure you can afford all these characters? 🤔

**Review by anaulin — APPROVED** [review]
✅  🍏

File: escape_roomba/thread_manager.py (hunk 1 above)
Comment (egnor) [reply escape_roomba/thread_manager.py:161]: 😛

**Pull request merged**